### PR TITLE
Russian translation

### PR DESCRIPTION
**Pull request by AAS**
100% translation.

Please, to keep the review as simple as possible, suggest only grammatical and mistranslation fixes.

Keep your upper/lower chars changes to other PRs.

### CODE REVIEW

**Comment by AJenbo** [comment]
ha ha, you just blew my mind :D

**Comment by AJenbo** [comment]
Would you be willing to also translate the 10 lines for the Android app?

https://github.com/diasurgical/devilutionX/blob/master/android-project/app/src/main/res/values/strings.xml
->
https://github.com/diasurgical/devilutionX/blob/master/android-project/app/src/main/res/values-ru/strings.xml

**Comment by AJenbo** [comment]
@Iniquitatis how do you feel about giving this a quick look?

@AAS I'll make sure to mark the 4 strings correctly

File: Translations/ru.po [hunk 1]
```diff
@@ -2456,7 +2476,7 @@ msgstr "Бесполезный"
 
 #: Source/itemdat.cpp:205
 msgid "Bent"
-msgstr "Кривой"
+msgstr "Согутый"
```
Comment (tsunamistate) [inline Translations/ru.po:2479]:
```suggestion
msgstr "Согнутый"
```

File: Translations/ru.po [hunk 2]
```diff
@@ -2408,43 +2428,43 @@ msgstr "Короткий посох заряженной стрелы"
...
 
 #: Source/itemdat.cpp:200
 msgid "Mithril"
-msgstr "Мирфрил"
+msgstr "Мирфриловый"
 
 #: Source/itemdat.cpp:201
 msgid "Meteoric"
-msgstr "Метиорит"
+msgstr "Метиоритный"
```
Comment (tsunamistate) [inline Translations/ru.po:2463]:
```suggestion
msgstr "Метеоритный"
```

File: Translations/ru.po [hunk 3]
```diff
@@ -2737,401 +2758,404 @@ msgstr "Кристаллический"
...
 
 #: Source/itemdat.cpp:356
 msgid "flame"
-msgstr "пламя"
+msgstr "пламени"
 
 #: Source/itemdat.cpp:357
 msgid "fire"
-msgstr "огонь"
+msgstr "огоня"
```
Comment (tsunamistate) [inline Translations/ru.po:3044]:
```suggestion
msgstr "огня"
```

File: Translations/ru.po [hunk 4]
```diff
@@ -3356,7 +3381,7 @@ msgstr "Шлем духов"
 
 #: Source/itemdat.cpp:460
 msgid "Thinking Cap"
-msgstr "Кепка мыслителя"
+msgstr "Кепка размышлений"
```
Comment (tsunamistate) [inline Translations/ru.po:3384]: "Кепка" вообще не нравится
Скорее "Шапка" подойдёт

File: Translations/ru.po [hunk 5]
```diff
@@ -3456,309 +3481,303 @@ msgstr "Сужающее кольцо"
...
-#| msgid "recover life and mana"
 msgid "restore some life and mana"
 msgstr "восстановить жизнь и ману"
 
 #: Source/items.cpp:2020
-#, fuzzy
-#| msgid "recover life and mana"
 msgid "restore all life and mana"
-msgstr "восстановить жизнь и ману"
+msgstr "восстановить жизнь и ману полнстью"
```
Comment (tsunamistate) [inline Translations/ru.po:3744]:
```suggestion
msgstr "полностью восстановить жизнь и ману"
```

File: Translations/ru.po [hunk 6]
```diff
@@ -4171,589 +4187,597 @@ msgstr "урон: {:d}-{:d}  Несокрушимый"
...
 
 #: Source/items.cpp:4223 Source/items.cpp:4232 Source/items.cpp:4239
 msgid "Not Identified"
-msgstr ""
+msgstr "Не идентифицировано"
 
 #: Source/loadsave.cpp:1665 Source/loadsave.cpp:2122
 msgid "Unable to open save file archive"
-msgstr ""
+msgstr "Невозможно открыть фал сохранённой игры"
```
Comment (tsunamistate) [inline Translations/ru.po:4226]:
```suggestion
msgstr "Невозможно открыть файл сохранённой игры"
```

File: Translations/ru.po [hunk 7]
```diff
@@ -4171,589 +4187,597 @@ msgstr "урон: {:d}-{:d}  Несокрушимый"
...
 #: Source/monstdat.cpp:98
 msgid "Maelstrom"
-msgstr ""
+msgstr "Вихрь"
 
+# ХЗ ХЗ
 #: Source/monstdat.cpp:99
 msgid "Devil Kin Brute"
-msgstr ""
+msgstr "Грубое дьявольское отродье"
```
Comment (tsunamistate) [inline Translations/ru.po:4523]: Diablo1.ru предлагает "Дьяволёнок"

File: Translations/ru.po [hunk 8]
```diff
@@ -4171,589 +4187,597 @@ msgstr "урон: {:d}-{:d}  Несокрушимый"
...
 
 #: Source/monstdat.cpp:103
 msgid "Death Wing"
-msgstr ""
+msgstr "Крыло смерти"
 
 #: Source/monstdat.cpp:104
 msgid "Slayer"
-msgstr ""
+msgstr "Убийца"
```
Comment (tsunamistate) [inline Translations/ru.po:4543]:
```suggestion
msgstr "Истребитель"
```

File: Translations/ru.po [hunk 9]
```diff
@@ -5736,195 +5763,195 @@ msgstr "Логово архиепископа Лазаря"
...
 
 #: Source/spelldat.cpp:24
 msgid "Infravision"
-msgstr ""
+msgstr "Инфравидение"
 
 #: Source/spelldat.cpp:25
 msgid "Phasing"
-msgstr ""
+msgstr "Премещение"
```
Comment (tsunamistate) [inline Translations/ru.po:5794]:
```suggestion
msgstr "Перемещение"
```

File: Translations/ru.po [hunk 10]
```diff
@@ -5736,195 +5763,195 @@ msgstr "Логово архиепископа Лазаря"
...
 
 #: Source/spelldat.cpp:46
 msgid "Holy Bolt"
-msgstr ""
+msgstr "Святой болт"
 
 #: Source/spelldat.cpp:47
 msgid "Resurrect"
-msgstr ""
+msgstr "Восресить"
```
Comment (tsunamistate) [inline Translations/ru.po:5870]:
```suggestion
msgstr "Воскресить"
```

**Review by tsunamistate — COMMENTED** [review]
Looks good!

Just needs some minor misspelling fixes.
I have also proposed some options for problematic translations

File: Translations/ru.po [hunk 11]
```diff
@@ -4171,589 +4187,597 @@ msgstr "урон: {:d}-{:d}  Несокрушимый"
...
 
 #: Source/monstdat.cpp:128 Source/spelldat.cpp:36
 msgid "Golem"
-msgstr ""
+msgstr "Голем"
 
 #: Source/monstdat.cpp:129
 msgid "The Dark Lord"
-msgstr ""
+msgstr "Темный Лорд"
```
Comment (Iniquitatis) [inline Translations/ru.po:4645]:
```suggestion
msgstr "Тёмный Лорд"
```

File: Translations/ru.po [hunk 12]
```diff
@@ -4171,589 +4187,597 @@ msgstr "урон: {:d}-{:d}  Несокрушимый"
...
 
 #: Source/monstdat.cpp:139
 msgid "Spider Lord"
-msgstr ""
+msgstr "Повелитель пауков"
 
 #: Source/monstdat.cpp:140
 msgid "Lashworm"
-msgstr ""
+msgstr "Плеточный червь"
```
Comment (Iniquitatis) [inline Translations/ru.po:4689]:
```suggestion
msgstr "Плёточный червь"
```

File: Translations/ru.po [hunk 13]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:487
 msgid "Soulpus"
-msgstr ""
+msgstr "Душегной"
 
 #: Source/monstdat.cpp:488
 msgid "Pukerat the Unclean"
-msgstr ""
+msgstr "Пукарат Нечистый"
```
Comment (Iniquitatis) [inline Translations/ru.po:4800]: Puke — блевать.
Блевокрыса? Звучит странно.
Что-то слишком много у Близзов блюющих тварей. Ahem.

File: Translations/ru.po [hunk 14]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:505
 msgid "Spineeater"
-msgstr ""
+msgstr "Спиногрыз"
 
 #: Source/monstdat.cpp:506
 msgid "Blackash the Burning"
-msgstr ""
+msgstr "Чёрнострел Пылающий"
```
Comment (Iniquitatis) [inline Translations/ru.po:4865]:
```suggestion
msgstr "Чернострел Пылающий"
```
Иначе два ударения в одном слове, либо одно неправильное.

File: Translations/ru.po [hunk 15]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:531
 msgid "Breakspine"
-msgstr ""
+msgstr "Костехруст"
 
 #: Source/monstdat.cpp:532
 msgid "Devilskull Sharpbone"
-msgstr ""
+msgstr "Череподявол Острокост"
```
Comment (Iniquitatis) [inline Translations/ru.po:4972]:
```suggestion
msgstr "Череподьявол Острокост"
```

File: Translations/ru.po [hunk 16]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:536
 msgid "Goldblight of the Flame"
-msgstr ""
+msgstr "Златомор Пламенный"
 
 #: Source/monstdat.cpp:537
 msgid "Blackstorm"
-msgstr ""
+msgstr "Чёрношторм"
```
Comment (Iniquitatis) [inline Translations/ru.po:4992]:
```suggestion
msgstr "Черношторм"
```

File: Translations/ru.po [hunk 17]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:544
 msgid "Lionskull the Bent"
-msgstr ""
+msgstr "Львочереп"
 
 #: Source/monstdat.cpp:545
 msgid "Blacktongue"
-msgstr ""
+msgstr "Чёрнояз"
```
Comment (Iniquitatis) [inline Translations/ru.po:5024]:
```suggestion
msgstr "Чернояз"
```

File: Translations/ru.po [hunk 18]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:548
 msgid "Fangskin"
-msgstr ""
+msgstr "Клыкошкур"
 
 #: Source/monstdat.cpp:549
 msgid "Witchfire the Unholy"
-msgstr ""
+msgstr "Огневедьм Нечистивый"
```
Comment (Iniquitatis) [inline Translations/ru.po:5040]:
```suggestion
msgstr "Огневедьм Нечестивый"
```

File: Translations/ru.po [hunk 19]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:550
 msgid "Blackskull"
-msgstr ""
+msgstr "Чёрночерен"
```
Comment (Iniquitatis) [inline Translations/ru.po:5044]:
```suggestion
msgstr "Черночереп"
```

File: Translations/ru.po [hunk 20]
```diff
@@ -4765,332 +4789,336 @@ msgstr "Костеголовый Кинакс"
...
 
 #: Source/monstdat.cpp:555
 msgid "Howlingire the Shade"
-msgstr ""
+msgstr "Ревущая Тень"
 
 #: Source/monstdat.cpp:556
 msgid "Doomcloud"
-msgstr ""
+msgstr "Судьбокров"
```
Comment (Iniquitatis) [inline Translations/ru.po:5068]: Судьбоблако!
...
Шутка?

File: Translations/ru.po [hunk 21]
```diff
@@ -5204,124 +5232,124 @@ msgstr "Игрок '{:s}' (уровень {:d}) уже в игре"
...
 
 #: Source/objects.cpp:120
 msgid "Quiet"
-msgstr ""
+msgstr "Тихий"
 
 #: Source/objects.cpp:121
 msgid "Secluded"
-msgstr ""
+msgstr "Уединенный"
```
Comment (Iniquitatis) [inline Translations/ru.po:5307]:
```suggestion
msgstr "Уединённый"
```

File: Translations/ru.po [hunk 22]
```diff
@@ -5563,46 +5588,49 @@ msgstr "{:s} (отключено)"
...
 
 #: Source/pfile.cpp:381
 msgid "Unable to open archive"
-msgstr ""
+msgstr "Невозможно открыть архив"
 
 #: Source/pfile.cpp:383
 msgid "Unable to load character"
-msgstr ""
+msgstr "Невозможно загрузить персонаж"
```
Comment (Iniquitatis) [inline Translations/ru.po:5599]:
```suggestion
msgstr "Невозможно загрузить персонажа"
```
Скорее, родительный падеж.

File: Translations/ru.po [hunk 23]
```diff
@@ -5563,46 +5588,49 @@ msgstr "{:s} (отключено)"
...
 
 #: Source/qol/monhealthbar.cpp:35 Source/qol/xpbar.cpp:56
 msgid ""
 "Failed to load UI resources.\n"
 "\n"
 "Make sure devilutionx.mpq is in the game folder and that it is up to date."
 msgstr ""
+"Не удалось загрузить ресурсы интерфейса.\n"
+"\n"
+"Убедитесь что devilutionx.mpq находится в папке с игрой и он обновлён."
```
Comment (Iniquitatis) [inline Translations/ru.po:5633]:
```suggestion
"Убедитесь, что devilutionx.mpq находится в папке с игрой и он обновлён."
```

File: Translations/ru.po [hunk 24]
```diff
@@ -5736,195 +5763,195 @@ msgstr "Логово архиепископа Лазаря"
...
 
 #: Source/spelldat.cpp:44
 msgid "Elemental"
-msgstr ""
+msgstr "Элементаль"
 
 #: Source/spelldat.cpp:45
 msgid "Charged Bolt"
-msgstr ""
+msgstr "Заряженый болт"
```
Comment (Iniquitatis) [inline Translations/ru.po:5862]:
```suggestion
msgstr "Заряженный болт"
```

File: Translations/ru.po (hunk 10 above)
Comment (Iniquitatis) [inline Translations/ru.po:5870]:
```suggestion
msgstr "Воскресить"
```

**Review by Iniquitatis — CHANGES_REQUESTED** [review]
Looks fine to me, aside from a few typos.

File: Translations/ru.po [hunk 25]
```diff
@@ -1266,217 +1265,219 @@ msgstr "Сохранение..."
...
 #: Source/error.cpp:55
 msgid "Wherever you go, there you are"
-msgstr ""
+msgstr "Куда бы вы ни пошли, вот и вы"
 
 #. TRANSLATORS: Shrine Text. Keep atmospheric. :)
 #: Source/error.cpp:56
 msgid "Energy comes at the cost of wisdom"
-msgstr ""
+msgstr "Энергия приходит за счет мудрости"
```
[the remark this answers is not in the document]
Comment (SourceCodeDeleted) [reply Translations/ru.po:1350]: Can our devikutionx habdle the ё letter? I think in most cases we can just leave it as е .

File: Translations/ru.po (hunk 1 above)
[the remark this answers is not in the document]
Comment (SourceCodeDeleted) [reply Translations/ru.po:2479]: I think согнутый is best here .

File: Translations/ru.po [hunk 26]
```diff
@@ -2737,401 +2758,404 @@ msgstr "Кристаллический"
...
 
 #: Source/itemdat.cpp:369
 msgid "vampires"
-msgstr "вампир"
+msgstr "вампиров"
 
 #: Source/itemdat.cpp:370
 msgid "the leech"
-msgstr "личь"
+msgstr "личя"
```
[the remark this answers is not in the document]
Comment (SourceCodeDeleted) [reply Translations/ru.po:3097]: Пьявка это leach но , мне меч пьявкы звучит странно )

File: Translations/ru.po (hunk 13 above)
Comment (SourceCodeDeleted) [reply Translations/ru.po:4800]: Чего не хочешь назвать его пукрат ?))

Может быть пьюкрат подходит лучше так как , это имя монстера ? 

Я об этом подумал чуть чуть и не могу придумать лучше. Блевокрыса хоть понятное

File: Translations/ru.po (hunk 20 above)
Comment (SourceCodeDeleted) [reply Translations/ru.po:5068]: Гммм это мне звучит странно .

File: Translations/ru.po (hunk 25 above)
[the remark this answers is not in the document]
Comment (Iniquitatis) [reply Translations/ru.po:1350]: @SourceCodeDeleted If that's the case, then we need to be consistent about it. There are already lots of occurrences of "ё" in the latest translation PRs.

File: Translations/ru.po (hunk 26 above)
[the remark this answers is not in the document]
Comment (Iniquitatis) [reply Translations/ru.po:3097]: Точно. Пропустил оригинал. `Lich` vs `leech`.

**Review by AAS — COMMENTED** [review]
> Would you be willing to also translate the 10 lines for the Android app?
> 
> https://github.com/diasurgical/devilutionX/blob/master/android-project/app/src/main/res/values/strings.xml
> ->
> https://github.com/diasurgical/devilutionX/blob/master/android-project/app/src/main/res/values-ru/strings.xml

Done.
https://github.com/diasurgical/devilutionX/pull/2481

File: Translations/ru.po (hunk 8 above)
Comment (tsunamistate) [reply Translations/ru.po:4543]: Окей, тогда оставим как есть

File: Translations/ru.po (hunk 20 above)
Comment (Iniquitatis) [reply Translations/ru.po:5068]: @AAS Да нет, это я просто оффтопом написал. =) Судбокров, кажется, уже звучит достаточно адекватно.

_Впрочем, я не шибко силён в литературной части и в названиях в частности._

File: Translations/ru.po (hunk 7 above)
Comment (Iniquitatis) [reply Translations/ru.po:4523]: @AAS По поводу `Brute`: возможно, `здоровяк` или `громила` (ссылаясь на [одно из значений](https://dictionary.cambridge.org/dictionary/english/brute)).

P.S. Кач**о**к. =)

Comment (SourceCodeDeleted) [reply Translations/ru.po:4523]: Мне нравится громила

**Comment by SourceCodeDeleted** [comment]
Когда я начал переводить все эти строчки , мне сложно было перевести эти благословения от шрайнов .

**Comment by SourceCodeDeleted** [comment]
Так как я не могу ответить прямо на все эти вопросы , 
Судьбомес - Я думаю что нормально .
Дьяволёнок  тоже хорошее , а дьяволёнок-качок мне странно звучит )

**Comment by AJenbo** [comment]
You guys will let me know when this is ready right? Having a hard time following the discussion 😅

**Comment by AAS** [comment]
@AJenbo All reported issues were fixed. It's done on my end (:

**Comment by AJenbo** [comment]
Thanks I'll see about the font later this week

File: Translations/ru.po (hunk 25 above)
[the remark this answers is not in the document]
Comment (SourceCodeDeleted) [reply Translations/ru.po]: @Iniquitatis  yes, I am looking at the pictures uploaded by @AJenbo  and I don't see the **Ё ё**  letters . @AJenbo  is there a E with two dots on it?

Besides, we can read without this letter )

**Comment by SourceCodeDeleted** [comment]
I just realised this is merged already .... sorry for notifications )

File: Translations/ru.po (hunk 25 above)
[the remark this answers is not in the document]
Comment (Iniquitatis) [reply Translations/ru.po]: While we can read without it, is there any reason for not adding it? Russian alphabet has more letters than English in any case.

**Comment by AJenbo** [comment]
@AAS @Iniquitatis @SourceCodeDeleted the Cyrillic font has not been implemented and all reported translation issues addressed. If you have time to update, test and complete the new parts of the translation that would be greatly appreciated as we are planning a release next month :)

image

For testing you can simply fetch the latest test build and put the .mo produced by Poedit in the game folder (set language in diablo.ini)

**Comment by AAS** [comment]
Oh, that looks awful in UI (:
I'll look into in it tomorrow or the day after.

**Comment by tsunamistate** [comment]
@AAS @Iniquitatis I have started a branch to unfuzzy and translate the text
https://github.com/tsunamistate/devilutionX/tree/update-ru-translation
Let's cooperate to avoid duplicating effort

**Comment by AAS** [comment]
@tsunamistate OK, I'll push my fixes into your repo.

**Comment by SourceCodeDeleted** [comment]
It seems that we probably should truncate some words somehow .

**Comment by SourceCodeDeleted** [comment]
> Oh, that looks awful in UI (:
> I'll look into in it tomorrow or the day after.

It seems that we probably should truncate some words somehow .

**Comment by AJenbo** [comment]
You can also break some of them in to two lines

**Comment by SourceCodeDeleted** [comment]
Sometimes when we want to shorten a word, we do this.

`сопротив.-е`

I think the - and having 
`сопро - 
тивление`

Might not look good. 

@AAS  what do you think?

**Comment by AAS** [comment]
@SourceCodeDeleted I've never seen anyone used `сопротив.-е` shortening style, it looks odd to me to put `.` before `-`, and we only 2 chars shorter than original word.
Also we can't go with `сопро - тивление` because there is no place for `молнии/огню/магии` in second line.
Right now it's `сопротив молнии/огню/магии` two liner  in @tsunamistate's branch.

**Comment by tsunamistate** [comment]
@AAS @SourceCodeDeleted made a draft PR based on current branch state of my fork: https://github.com/diasurgical/devilutionX/pull/2970
So let's continue discussion there

**Pull request merged**